l1-contracts/contracts/upgrades/UpgradeHyperchains.sol

```diff
@@ -0,0 +1,33 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.24;
+
+import {Diamond} from "../state-transition/libraries/Diamond.sol";
+import {BaseZkSyncUpgrade, ProposedUpgrade} from "./BaseZkSyncUpgrade.sol";
+
+/// @author Matter Labs
+/// @custom:security-contact security@matterlabs.dev
+contract UpgradeHyperchains is BaseZkSyncUpgrade {
+    /// @notice The main function that will be called by the upgrade proxy.
+    /// @param _proposedUpgrade The upgrade to be executed.
+    function upgradeWithAdditionalData(
+        ProposedUpgrade calldata _proposedUpgrade,
```

Comment:
`ProposedUpgrade` contains the `postUpgradeCalldata`. You may use it and then you may leave the same function selector and reuse the tool that constructs the upgrade calldata.
- https://github.com/matter-labs/era-contracts/blob/dev/l1-contracts/contracts/upgrades/BaseZkSyncUpgrade.sol#L35